File: tasks/prac4_js/task6.js
const prompt = require('prompt-sync')();

const password = prompt("Введите пароль: ");

const hasLength = password.length >= 8;
const hasDigit = /\d/.test(password);
const hasUpper = /[A-Z]/.test(password);

if (!hasLength) {
    console.log("Ошибка: пароль должен содержать не менее 8 символов");
} else if (!hasDigit) {
    console.log("Ошибка: пароль должен содержать хотя бы одну цифру");
} else if (!hasUpper) {
    console.log("Ошибка: пароль должен содержать хотя бы одну заглавную букву");
} else {
    console.log("Пароль надежный!");
}

File: tasks/prac4_python/task6.py
password = input("Введите пароль: ")

has_length = len(password) >= 8
has_digit = any(char.isdigit() for char in password)
has_upper = any(char.isupper() for char in password)

# https://docs-python.ru/tutorial/vstroennye-funktsii-interpretatora-python/funktsija-len/
# https://docs-python.ru/tutorial/operatsii-tekstovymi-strokami-str-python/metod-str-isdigit/
# https://docs-python.ru/tutorial/operatsii-tekstovymi-strokami-str-python/metod-str-isupper/

if not has_length:
    print("Ошибка: пароль должен содержать не менее 8 символов")
elif not has_digit:
    print("Ошибка: пароль должен содержать хотя бы одну цифру")
elif not has_upper:
    print("Ошибка: пароль должен содержать хотя бы одну заглавную букву")
else:
    print("Пароль надежный!")
